app/models/manageiq/providers/embedded_terraform/automation_manager/azure_credential.rb

```diff
@@ -0,0 +1,80 @@
+class ManageIQ::Providers::EmbeddedTerraform::AutomationManager::AzureCredential < ManageIQ::Providers::EmbeddedTerraform::AutomationManager::CloudCredential
+  COMMON_ATTRIBUTES = [].freeze
+
+  EXTRA_ATTRIBUTES = [
+    {
+      :component  => 'text-field',
+      :label      => N_('Subscription ID'),
+      :helperText => N_('The Subscription ID for the Microsoft Azure account'),
+      :name       => 'subscription',
+      :id         => 'subscription',
+      :isRequired => true,
+      :validate   => [{:type => 'required'}],
+    },
+    {
+      :component  => 'text-field',
+      :label      => N_('Tenant ID'),
+      :helperText => N_('The Tenant ID for the Microsoft Azure account'),
+      :name       => 'tenant',
+      :id         => 'tenant',
+      :maxLength  => 1024,
+    },
+    {
+      :component  => 'password-field',
+      :label      => N_('Client Secret'),
+      :helperText => N_('The Client Secret for the Microsoft Azure account'),
+      :name       => 'secret',
+      :id         => 'secret',
+      :type       => 'password',
+      :maxLength  => 1024,
+    },
+    {
+      :component  => 'text-field',
+      :label      => N_('Client ID'),
+      :helperText => N_('The Client ID for the Microsoft Azure account'),
+      :name       => 'client',
+      :id         => 'client',
+      :maxLength  => 128,
+    },
+  ].freeze
+
+  API_ATTRIBUTES = (COMMON_ATTRIBUTES + EXTRA_ATTRIBUTES).freeze
+
+  API_OPTIONS = {
+    :type       => 'cloud',
+    :label      => N_('Azure'),
+    :attributes => API_ATTRIBUTES
+  }.freeze
+
+  alias secret auth_key
+
+  def self.display_name(number = 1)
+    n_('Credential (Microsoft Azure)', 'Credentials (Microsoft Azure)', number)
+  end
+
+  def self.params_to_attributes(params)
+    attrs            = super.dup
+    attrs[:auth_key] = attrs.delete(:secret) if attrs.key?(:secret)
+
+    if %i[client tenant subscription].any? { |opt| attrs.has_key?(opt) }
+      attrs[:options]              ||= {}
+      attrs[:options][:client]       = attrs.delete(:client)       if attrs.key?(:client)
+      attrs[:options][:tenant]       = attrs.delete(:tenant)       if attrs.key?(:tenant)
+      attrs[:options][:subscription] = attrs.delete(:subscription) if attrs.key?(:subscription)
+    end
+
+    attrs
+  end
+
+  def client
+    options && options[:client]
+  end
+
+  def tenant
+    options && options[:tenant]
```

Comment:
I know this is in connection to the azure tenant, but since we also have a tenant concept in manageiq, I wonder if it makes sense to be explicit that this is the cloud_provider_tenant.  In this case, I think it's a name and not an id/number.

Reply:
Yeah that's a good point, in the provider code we call this a "cloud_tenant"

Reply:
Again this was taken from ansible implementation -> https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/embedded_ansible/automation_manager/azure_credential.rb#L89

The azure 'tenant' is available in `options`.
Will remove `def tenant` ,  can't say about ansible,  who needs separate method for terraform anyways ?

Reply:
Anyways, I replaced 'tenant' with `provider_tenant`